revolutionpi/digital_interrupts.go

```diff
@@ -25,10 +25,15 @@ type digitalInterrupt struct {
 	outputOffset     uint16
 	inputOffset      uint16
 	enabled          bool
+	isEncoder        bool
 	interruptAddress uint16
 }
 
-func initializeDigitalInterrupt(pin SPIVariable, g *gpioChip) (*digitalInterrupt, error) {
+type diWrapper struct {
+	pin *digitalInterrupt
+}
```

Comment:
encoders and digital interrupts are configured the same way in the revolution pi, so I tried to setup the code in a way to share the same initialization code. not sure if this is the best way to do it but it appears to be working. 

I had initially tried to make `digitalInterrupt` implement the encoder interface at first, but unfortunately, our interrupts interface and the resource interface both have a `Name` function, and go doesn't let you overload.

Reply:
This github comment is excellent! Make it a comment in the code, too, or else someone is going to mistakenly try to simplify this in 2025.